src/Discovery/test/Eureka.Test/EurekaHealthCheckHandlerTest.cs

```diff
@@ -23,10 +25,10 @@ public sealed class EurekaHealthCheckHandlerTest
     [Theory]
     [InlineData(HealthStatus.Down, InstanceStatus.Down)]
     [InlineData(HealthStatus.Up, InstanceStatus.Up)]
-    [InlineData(HealthStatus.Warning, InstanceStatus.Unknown)]
+    [InlineData(HealthStatus.Warning, InstanceStatus.Up)]
     [InlineData(HealthStatus.Unknown, InstanceStatus.Unknown)]
     [InlineData(HealthStatus.OutOfService, InstanceStatus.OutOfService)]
-    [InlineData(null, InstanceStatus.Unknown)]
+    [InlineData(null, InstanceStatus.Up)]
```

Comment:
I have an initial reaction of disliking this change, as part of me feels like a null health check _should_ result in an Unknown instance status. It looks like you are thinking that for this purpose, Steeltoe should treat `null` as erroneous and ignore it, is that accurate?

Reply:
`IHealthContributor` returns `null` to indicate the check is currently turned off. When _all_ registered checks are turned off, the aggregate health status is effectively unavailable and should not affect the instance status. I suppose this should be changed to `[InlineData(null, null)]` and have the caller deal with that case.

`IHealthContributor` returns `Unknown` when the health check can't be performed. Possible causes:
- An unhandled exception during evaluation
- The system is misconfigured
- Some information isn't available (yet) *1)

*1) This is the tricky one. Normally, health checks execute as part of an HTTP request, _after_ the app has fully started. For service discovery, however, it starts running health checks from a background thread as part of `EurekaDiscoveryClient` instantiation (so possibly _before_ the app startup has completed, depending on configured timings).
- `LivenessHealthContributor` and `ReadinessHealthContributor` initialize during `IHostApplicationLifetime.ApplicationStarted`, returning `Unknown` when called before that happened.
- `EurekaServerHealthContributor` outcome depends on:
  - The remote instance status (which hasn't been fetched yet)
  - Last applications fetch status (which hasn't happened yet)
  - Heartbeat status (which is currently executing, so there's no outcome yet).

  The caller being the heartbeat thread, none of these are available, so it returns `Unknown`.

One thing we can do is not run health checks when the instance status is `Starting`. That solves the problem in case `IsInstanceEnabledOnInit` is configured as `false` and the user marks when startup has completed. However, this doesn't solve the default configuration (`false`). In that case, we could run our own hosted service, signaling when it's okay to run health checks. With those in place, we can translate health status `Unknown` to instance status `Unknown`.

What do you think? I'm open to suggestions.

Reply:
We've discussed this and landed on the plan to translate health outcome `Unknown` to instance status `Up` (so the load balancer keeps sending traffic to it), but log a warning. However, today I realized that doesn't solve the `EurekaServerHealthContributor` chicken-and-egg problem.

What I ultimately ended up with, is ignoring the _first_ execution of `EurekaServerHealthContributor`. On later executions, an earlier heartbeat must have completed (succeeded or failed).

The `LivenessHealthContributor`/`ReadinessHealthContributor` taking the instance out of rotation when the app hasn't been started yet is a good thing (don't add them if you don't want to postpone traffic to it until fully started), so it doesn't need to be special-cased.